packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -107,12 +116,23 @@ export default function vercelServerless({
 	return {
 		name: PACKAGE_NAME,
 		hooks: {
-			'astro:config:setup': ({ command, config, updateConfig, injectScript }) => {
-				if (command === 'build' && analytics) {
-					injectScript('page', 'import "@astrojs/vercel/analytics"');
+			'astro:config:setup': async ({ command, config, updateConfig, injectScript }) => {
+				if (webAnalytics?.enabled) {
+					injectScript(
+						'page',
+						await getInjectableWebAnalyticsContent(
+							{
+								...webAnalytics.config,
+								mode: command === 'dev' ? 'development' : 'production',
+							},
+							config.root
+						)
+					);
+				}
+				if (command === 'build' && speedInsights?.enabled) {
+					injectScript('page', 'import "@astrojs/vercel/speed-insights"');
```

Comment:
```suggestion
					injectScript('page', 'import "@astrojs/vercel/speed-insights.js"');
```

Extension is important in an ESM environment.

Reply:
Oh, it's in the `package.json`! I missed that. So, in practice, it should be fine as is. :) You can ignore my suggestion.